libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/map/MapLayoutCoordinator.kt

```diff
@@ -27,57 +24,37 @@ internal class MapLayoutCoordinator(
     private val binding: MapboxNavigationViewLayoutBinding
 ) : UICoordinator<ViewGroup>(binding.mapViewLayout) {
 
-    private val viewGroup = binding.mapViewLayout
-    private val mapViewOwner = navigationViewContext.mapViewOwner
+    private var loadStyleJob: Job? = null
     private val mapStyleLoader = navigationViewContext.mapStyleLoader
-    private var reloadStyleJob: Job? = null
 
     override fun onDetached(mapboxNavigation: MapboxNavigation) {
         super.onDetached(mapboxNavigation)
         mapStyleLoader.mapboxMap = null
     }
 
     override fun MapboxNavigation.flowViewBinders(): Flow<Binder<ViewGroup>> {
-        return navigationViewContext.mapView
-            .map { mapViewOverride ->
-                if (mapViewOverride == null) {
-                    Scene.getSceneForLayout(
-                        viewGroup,
-                        R.layout.mapbox_mapview_layout,
-                        viewGroup.context,
-                    ).enter()
-
-                    val mapView = BoundMapViewProvider.bindLayoutAndGet(viewGroup)
-                    initDefaultMap(mapView.getMapboxMap())
-                    mapViewOwner.updateMapView(mapView)
-                    mapView
-                } else {
-                    initCustomMap(mapViewOverride.getMapboxMap())
-                    viewGroup.removeAllViews()
-                    viewGroup.addView(mapViewOverride)
-                    mapViewOwner.updateMapView(mapViewOverride)
-                    mapViewOverride
+        return navigationViewContext.uiBinders.mapViewBinder.map { customBinder ->
+            loadStyleJob?.cancel()
+            loadStyleJob = null
+            (customBinder ?: MapboxMapViewBinder()).also {
+                it.context = navigationViewContext
+                it.navigationViewBinding = binding
+                it.registerMapboxMapObserver { map ->
+                    mapStyleLoader.mapboxMap = map
+                }
+                when (it.getMapStyleLoadPolicy()) {
+                    MapStyleLoadPolicy.ONCE -> mapStyleLoader.loadInitialStyle()
+                    MapStyleLoadPolicy.ON_CONFIGURATION_CHANGE -> {
+                        mapStyleLoader.loadInitialStyle()
+                        loadStyleJob = coroutineScope.launch {
+                            mapStyleLoader.observeAndReloadNewStyles()
+                        }
+                    }
+                    else -> {
+                        // do nothing
+                    }
```

Comment:
I am afraid that this won't work as expected. The code inside this `.also` function block will execute before `mapStyleLoader.mapboxMap = map`. 

Notice that `UICoordinator` calls `bind` when collecting this flowable, while this code runs immediately after Binder is created. All calls to `mapStyleLoader.loadInitialStyle()` will either do nothing or reloads initial style into previous assigned MapView instance.

I think we can adjust that behaviour (and remove the redundant `MapboxMapObserver`) by utilizing the existing `MapViewOwner`. By introducing an extension method that run function block when new MapView is attached:    
```kotlin
@ExperimentalPreviewMapboxNavigationAPI
internal inline fun MapViewOwner.doOnAttachMapView(crossinline action: (MapView) -> Unit) {
    registerObserver(object: MapViewObserver() {
        override fun onAttached(mapView: MapView) {
            action(mapView)
            unregisterObserver(this)
        }
    })
}
```
we could do something like this:
```kotlin
override fun MapboxNavigation.flowViewBinders(): Flow<Binder<ViewGroup>> {
    return navigationViewContext.uiBinders.mapViewBinder.map { customBinder ->
        loadStyleJob?.cancel()
        loadStyleJob = null
        (customBinder ?: MapboxMapViewBinder()).also {
            it.context = navigationViewContext
            it.navigationViewBinding = binding
            navigationViewContext.mapViewOwner.doOnAttachMapView { mapView ->
                mapStyleLoader.mapboxMap = mapView.getMapboxMap()
                when (it.getMapStyleLoadPolicy()) {
                    MapStyleLoadPolicy.ONCE -> mapStyleLoader.loadInitialStyle()
                    MapStyleLoadPolicy.ON_CONFIGURATION_CHANGE -> {
                        mapStyleLoader.loadInitialStyle()
                        loadStyleJob = coroutineScope.launch {
                            mapStyleLoader.observeAndReloadNewStyles()
                        }
                    }
                    else -> {
                        // do nothing
                    }
                }
            }
        }
    }
}
```

Reply:
> The code inside this .also function block will execute before mapStyleLoader.mapboxMap = map.

Hm, I thought I could load styles before setting the map. But I see in the code that I can't. Fixed.